src/Sylius/Component/Core/Exception/InvalidTransitionException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Exception;
+
+use SM\SMException;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class InvalidTransitionException extends SMException
+{
+    /**
+     * @param string $transition
+     */
+    public function __construct($transition)
```

Comment:
DX: Can we also pass an optional list of available states? Or whole `StateMachine` and build even more helpful message? "Transition foo is invalid for bar state machine. Available states are: rynkowski, krawczyk, hołdys"?